CONTRIBUTING.md

```diff
@@ -0,0 +1,190 @@
+   the username `admin` and the password you set up.
+3. To access the Splunk [REST API][CS2],
+   you can make API calls from outside the container using the `curl` command.
+   For example, you can run the following command to search all data:
+     ```
+     curl -u admin:YourPassword -k https://localhost:8089/services/search/jobs -d search="search *"
+     ```
+4. If you want to use the Splunk UI, open a web browser on the host and navigate to
+   `localhost:8000`.
+
+[CS1]:
+https://docs.splunk.com/Documentation/Splunk/9.0.4/Installation/DeployandrunSplunkEnterpriseinsideDockercontainers
+[CS2]:
+https://docs.splunk.com/Documentation/Splunk/9.0.4/RESTTUT/RESTTutorialIntro
+
+### Building and running the segment-bridge container image
+
+The scripts in this repo can be built into a container image to enable
+scheduling and running them on K8s clusters.
+
+To build the image locally, one needs to be logged in to a `redhat.com` account
+(With e.g `podman login`) in order to access the base image and then the image
+can be built with:
+```
+podman build -t segment-bridge .
+```
+
+The scripts require access to Splunk, Segment and OpenShift credentials. One
+way to provide such access is to mount the local `~/.netrc` and
+`~/.kube/config` files (Assuming they contain suitable credentials) to the
+image with a command like the following:
+```
+podman run -it --rm \
+         -v ~/.netrc:/usr/local/etc/netrc:z \
+         -v ~/.kube/config:/usr/local/etc/kube_config:z \
+         segment-bridge
+```
+The following command can be run inside the container to test the full chain of
+scripts. This will copy real data from the staging audit logs in Splunk into the
+Segment DEV environment:
+```
+fetch-uj-records.sh | splunk-to-segment.sh | segment-mass-uploader.sh
+```
+
+### Unit Tests
+Go unit tests are included in various packages within the repository.
+Go unit tests are located within the tests directory, with filenames ending with
+_tests.go and with .go.
+
+#### Running the Unit Tests Locally
+1. Clone your fork of the project.
+2. Navigate to the project's root directory
+3. To run all the Go unit tests in the repository,
+execute the following command `go test ./...`
+
+```
+?   	github.com/redhat-appstudio/segment-bridge.git/cmd/querygen	[no test files]
+ok  	github.com/redhat-appstudio/segment-bridge.git/querygen	0.002s
+ok  	github.com/redhat-appstudio/segment-bridge.git/queryprint	(cached)
+ok  	github.com/redhat-appstudio/segment-bridge.git/scripts	0.002s
+ok  	github.com/redhat-appstudio/segment-bridge.git/segment	(cached)
+ok  	github.com/redhat-appstudio/segment-bridge.git/stats	(cached)
+ok  	github.com/redhat-appstudio/segment-bridge.git/webfixture	(cached)
+```
+4. If you want to run tests for a specific package, you can do so by providing
+the package path, like this:
+```
+go test ./querygen
+```
+
+#### Test Coverage
+[TBD]
+
+### Integration Tests
+[TBD]
+
+#### Running the Integration Tests
+[TBD]
+
+### Commit Messages
```

Comment:
please mention somewhere that one should run gofmt and all the unit tests before sending a PR.

Reply:
Done